src/UI/Art/SpriteArtist/PlayerSetupColorFactionArtist.java

```diff
@@ -127,25 +141,47 @@ private static void initialize()
     gridHeight = numFactions*unitSizePx + unitBuffer*(numFactions-1);
   }
 
-  private static int nextFac = 0;
-  private static int nextCol = 0;
-  private static boolean donePreloading = false;
-  /** This can be called repeatedly on the sly to get all infantry sprites
-   * in memory and make this option screen less sluggish on first entry. */
-  public static boolean preloadOneInfantrySprite(String unitName)
+  public static class InfantrySpritePreloader implements Runnable
   {
-    if( !donePreloading )
+    private Thread myThread;
+    private String myUnitName;
+    private int nextFac = 0;
+    private int nextCol = 0;
+
+    public InfantrySpritePreloader(String unitName)
     {
-      SpriteLibrary.getMapUnitSpriteSet(unitName, UIUtils.getFactions()[nextFac], UIUtils.getCOColors()[nextCol]);
-      nextFac++;
-      if( nextFac >= UIUtils.getFactions().length )
+      myThread = new Thread(this);
+      myThread.start();
+      myUnitName = unitName;
+    }
+
+    @Override
+    public void run()
+    {
+      while( !donePreloading )
```

Comment:
Kind of a weird loop structure. Why not just use a nested for?

Reply:
I just copied/moved what had been done before. A simple double loop would be more straightforward though.